### PR TITLE
 Fix: Travis builds aren't working anymore (#62) 

### PR DESCRIPTION
**Pull request by achave11-ucsc**
https://github.com/DataBiosphere/hca-metadata-api/issues/62

### CODE REVIEW

**Review by dsotirho-ucsc — APPROVED** [review]
Approved.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    - name: Test
+      run: |
+        python3.6 -m venv .venv
+        source .venv/bin/activate
+        make travis_install
+        # Hack: The use of chrgp compensates for a quirk of Docker. The PyCharm image
+        # used by make format sets up a user called `developer` and assigns it UID
+        # 1000. Travis is running as UID 2000. An alternative would be to pass --user
+        # to `docker run` and bind-mount an /etc/passwd that maps that to `developer`.
+        # Currently I'm unclear why `make format` works unchanged on Docker
```
Comment (hannes-ucsc) [inline .github/workflows/ci.yml]:
```suggestion
        # Hack: The use of `chrgp` compensates for a quirk of Docker. The PyCharm image
        # used by `make format` sets up a user called `developer` and assigns it UID
        # 1000. Travis is running as UID 2000. An alternative would be to pass --user
        # to `docker run` and bind-mount an /etc/passwd that maps that to `developer`.
        # Currently, we're unclear why `make format` works as is on developer machines.
```

The reference to Travis is stale. And we should check if the claim for Travis still applies to GH Action, and if this hack is even necessary for GH Actions.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (achave11-ucsc) [reply .github/workflows/ci.yml]: It seems like the claim is still valid for GH Actions but with a different UID, removing the hack [caused issues](https://github.com/DataBiosphere/hca-metadata-api/runs/4973363065?check_suite_focus=true#step:5:176).

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        restore-keys: |
+          ${{ runner.os }}-pip-
+    - name: Test
+      run: |
+        python3.6 -m venv .venv
+        source .venv/bin/activate
+        make travis_install
+        # Hack: The use of `chrgp` compensates for a quirk of Docker. The PyCharm image
+        # used by `make format` sets up a user called `developer` and assigns it UID
+        # 1000. GitHub Action runs as UID 121. An alternative would be to pass --user
```
Comment (hannes-ucsc) [inline .github/workflows/ci.yml]:
```suggestion
        # 1000. GitHub Actions runs as UID 121. An alternative would be to pass --user
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+          python-version: ${{ env.python_version }}
+      - name: Download coverage file
+        uses: actions/download-artifact@v2
+        with:
+          name: coverage-file
+      - name: Coveralls
+        run: |
+          pip install coveralls==3.0.1
+          # This is the token associated with the repository and only GH users
+          # with admin privileges would be able to retrive the token.
```
Comment (hannes-ucsc) [inline .github/workflows/ci.yml]:
```suggestion
          # This API token is associated with the repository, not any particular user. A GitHub user
          # with admin privileges must retrieve the token from
          #
          # https://coveralls.io/github/DataBiosphere/hca-metadata-api
          # 
          # and deposit it in
          #
          # https://github.com/DataBiosphere/hca-metadata-api/settings/secrets/actions
          #
          # as a secret of the name `COVERALLS_REPO_TOKEN`.
```

Rewrap

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        with:
+          name: coverage-file
+      - name: Coveralls
+        run: |
+          pip install coveralls==3.0.1
+          # This API token is associated with the repository, not any particular user. A 
+          # GitHub user with admin privileges must retrieve the token from 
+          # https://coveralls.io/github/DataBiosphere/hca-metadata-api and deposit it in
+          # https://github.com/DataBiosphere/hca-metadata-api/settings/secrets/actions
+          # as a secret of the name `COVERALLS_REPO_TOKEN`.
```
Comment (hannes-ucsc) [inline .github/workflows/ci.yml]:
```suggestion
          # as a secret called `COVERALLS_REPO_TOKEN`.
```

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
I just enabled PR comments by Coveralls. To test this, I'm requesting a cosmetic change.

image

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
Still no coverage comment from Coveralls. 

Also, the link looks weird:

image

It doesn't look like that on Azul (https://coveralls.io/repos/301780/build) and on Azul there is only one build per PR (the one for the simulated merge commit). Here there are two, one for the simulated merge commit and one for the HEAD of the PR branch.

I've just now set format to compact, like they are on Azul.

image

**Comment by theathorn** [comment]
@hannes-ucsc :"I will create a follow-on ticket about the missing coverage comment and review this again. After that., I will un-hold and review Daniel's PR #63 as well."

**Review by hannes-ucsc — APPROVED** [review]

**Pull request merged**